pkg/datasource.go

```diff
@@ -72,53 +98,71 @@ type GrafanaSearchRequest struct {
 type GrafanaCommonRequest struct {
 	Compartment string
 	Environment string
+	TenancyMode string
 	QueryType   string
 	Region      string
+	Tenancy     string // the actual tenancy with the format <configfile entry name/tenancyOCID>
 	TenancyOCID string `json:"tenancyOCID"`
 }
 
 // Query - Determine what kind of query we're making
 func (o *OCIDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
 	var ts GrafanaCommonRequest
+	var takey string
 
 	query := req.Queries[0]
 	if err := json.Unmarshal(query.JSON, &ts); err != nil {
 		return &backend.QueryDataResponse{}, err
 	}
 
 	queryType := ts.QueryType
-	if o.config == nil {
-		configProvider, err := getConfigProvider(ts.Environment)
+
+	o.logger.Debug("QueryData")
```

Comment:
The o.config here is ripped off because the function is now provided by the getConfigProvider (line 320) new function. There are several debug messages, for each of the error condition there. I do not think we miss any debug message there.

Reply:
In the new function, for example we have this error handling: `				var configProvider common.ConfigurationProvider
				configProvider = common.CustomProfileConfigProvider(oci_config_file, key)
				metricsClient, err := monitoring.NewMonitoringClientWithConfigurationProvider(configProvider)
				if err != nil {
					o.logger.Error("Error with config:" + key)
					return errors.New(fmt.Sprint("error with client", spew.Sdump(configProvider), err.Error()))
				}`